src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs

```diff
@@ -156,6 +158,57 @@ private CentralPackageVersion ToCentralPackageVersion(string packageId, string v
             return new CentralPackageVersion(packageId, VersionRange.Parse(version));
         }
 
+        private RestoreAuditProperties GetRestoreAuditProperties()
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            string enableAudit = _vsProjectAdapter.BuildProperties.GetPropertyValue(ProjectBuildProperties.NuGetAudit);
+            string auditLevel = _vsProjectAdapter.BuildProperties.GetPropertyValue(ProjectBuildProperties.NuGetAuditLevel);
+            string auditMode = _vsProjectAdapter.BuildProperties.GetPropertyValue(ProjectBuildProperties.NuGetAuditMode);
+            HashSet<string> suppressedAdvisories = GetSuppressedAdvisories();
+
+            return new RestoreAuditProperties()
+            {
+                EnableAudit = enableAudit,
+                AuditLevel = auditLevel,
+                AuditMode = auditMode,
+                SuppressedAdvisories = suppressedAdvisories,
+            };
+        }
+
+        private HashSet<string> GetSuppressedAdvisories()
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            IEnumerable<(string ItemId, string[] ItemMetadata)> buildItems = _vsProjectAdapter.GetBuildItemInformation(ProjectBuildProperties.NuGetAuditSuppress);
+            if (buildItems is null)
+            {
+                return null;
+            }
+            else if (buildItems is ICollection<(string, string[])> collection)
+            {
+                if (collection.Count == 0) return null;
+
+                var suppressedAdvisories = new HashSet<string>(collection.Count, StringComparer.OrdinalIgnoreCase);
+                foreach ((string itemId, _) in buildItems.NoAllocEnumerate())
+                {
+                    suppressedAdvisories.Add(itemId);
+                }
+
+                return suppressedAdvisories;
+            }
+            else
+            {
+                var suppressedAdvisories = new HashSet<string>(StringComparer.Ordinal);
```

Comment:
Should this also StringComparer.OrdinalIgnoreCase just like line no. 192?

Reply:
Yeah nice catch. 

Fortunately, I don't expect the 3rd tree to ever be hit :D

Reply:
(unresolved the conversation) If we don't expect the 3rd tree to ever be executed, I am wondering if we really need that code. I don't want team members in the future to think that every time we read MSBuild items, we should consider this as a pattern. Please correct if I am missing something.

Reply:
There's a chance the type returned is not ICollection (IEnumerable), so by inheritance the 3rd branch is necessary. 
We just know there's 1 implementation of that service and we don't expect there to be another one, but that's not a assumption our code can make safely.

Reply:
Thanks for clarifying.